### PR TITLE
Import access object

### PR DESCRIPTION
**Pull request by tushar-compro**

### CODE REVIEW

File: server/libs/skills/ac/common/acSkill.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
+const BaseSkill = require("../../common/baseSkill");
+//Access based Common Functionality goes here 
+
+const xmlUtil = require("../../../../utils/xmlUtil");
+
+module.exports = class AccessBaseSkill extends BaseSkill {
+//tooltip text to be displayed
+  getTootlTipText(skillParams) {
```
Comment (tushar-compro) [inline server/libs/skills/ac/common/acSkill.js]: getTaskbarTooltipText

File: server/libs/skills/ac/common/acSkill.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    var paramValueObj = skillParams.paramsObj;
+    var resolveParams = { "attrValue":  "Access - " + paramValueObj["DocTitle"] };
+    return Promise.resolve(resolveParams);
+  }
+   getSelectedObject(skillParams){
```
Comment (tushar-compro) [inline server/libs/skills/ac/common/acSkill.js]: getSelectedDBObjectName

File: server/libs/skills/ac/common/acSkill.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+    while (navaigationPaneDatabaseObjectArray.value.length > 0) {
+      navaigationPaneDatabaseObjectArray.value.pop(); //https://jsperf.com/array-clear-methods/3
+    }
+
+    for (let key in navigationPaneJSON) {
+      if (navigationPaneJSON[key]) {
+        objectType = objectMap[key];
+        for(let i=0; i<navigationPaneJSON[key].length;i++ ){
+            navaigationPaneDatabaseObjectArray.value.push({"label": (objectType + ' : ' + navigationPaneJSON[key][i].name),"data":(objectType + '.' + navigationPaneJSON[key][i].name)});
```
Comment (vsingb3) [inline server/libs/skills/ac/common/acSkill.js]: ObjectType: ObjectName

data = {
objectType: "",
objectName: ""
}

File: server/libs/skills/ac/importAccessObject/importAccessObject.js [hunk 4]
```diff
@@ -0,0 +1,175 @@
+// class importAccessobject contains all the functions for the IMPORT ACCESS OBJECT with the object implementation of the parameterArray
+const AccessBaseSkill = require("../common/acSkill");
+
+class importAccessobject extends AccessBaseSkill {
+  constructor(){
+    super();
+    this.projJSON = {};
+    this.IntialDataJSON ={};
```
Comment (vsingb3) [inline server/libs/skills/ac/importAccessObject/importAccessObject.js]: importDialogInputJSON
importedObjectJSON

File: server/libs/skills/ac/importAccessObject/importAccessObject.js [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+    var paramValueObj = skillParams.paramsObj;
+    var resolveParams = { "attrValue":paramValueObj.skillmode};
+    return Promise.resolve(resolveParams);
+  }
+  //This function return a string to be used for an atrribute of Get External Data to demonstrate the name with which the nam eof the step have to be saved.
+  getSaveAsText(skillParams){
+    var paramValueObj = skillParams.paramsObj;
+    var fileName = this.splitFilename(paramValueObj.databasePath);
+    //this code replaces the extension with empty string, effectively removing the File Extension
+    var resolveParams = fileName.replace("."+fileName.split(".")[fileName.split(".").length-1],"")
```
Comment (vsingb3) [inline server/libs/skills/ac/importAccessObject/importAccessObject.js]: var temp = fileName.split(".");
var fileExt = "." + temp[temp.length - 1]
filename = fileName.replace(fileExt, "");

File: server/libs/skills/ac/importAccessObject/importAccessObject.js [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+    // splitting in "." might end the file name before it gets completed -done
+    resolveParams = { "attrValue":"Import-"+resolveParams};
+    return Promise.resolve(resolveParams);
+  }
+  //This function splits the document name and returns the access database file name.
+  splitFilename(filePath){
```
Comment (vsingb3) [inline server/libs/skills/ac/importAccessObject/importAccessObject.js]: getFileNameFromFilePath

**Review by vsingb3 — APPROVED** [review]

**Pull request merged**